### ISSUE TITLE
I'm trying to integrating Wan Vace 14B into DiffSynth-Studio, but getting terrible results.

### ISSUE THREAD

**Issue by matabear-wyx** [modelscope/DiffSynth-Studio#582]
Model config:
{
  "dim": 5120,
  "eps": 1e-06,
  "ffn_dim": 13824,
  "freq_dim": 256,
  "in_dim": 16,
  "model_type": "vace",
  "num_heads": 40,
  "num_layers": 40,
  "out_dim": 16,
  "text_len": 512,
  "vace_layers": [0, 5, 10, 15, 20, 25, 30, 35],
  "vace_in_dim": 96
}

Inference scripts:

`import sys
import os
sys.path.append(os.path.abspath(os.path.join(os.path.dirname(__file__), '../../')))
# from src.wan_video_pipe import WanVideoPipeline
import torch
from diffsynth import ModelManager, save_video, VideoData, WanVideoPipeline
from diffsynth.models.wan_video_vace import VaceWanModel
from diffsynth.models.wan_video_dit import WanModel
from diffsynth.models.wan_video_text_encoder import WanTextEncoder
from diffsynth.models.wan_video_image_encoder import WanImageEncoder
from diffsynth.models.wan_video_vae import WanVideoVAE
from diffsynth.models.model_manager import ModelDetectorFromSingleFile, ModelDetectorFromSplitedSingleFile, ModelDetectorFromHuggingfaceFolder, ModelDetectorFromPatchedSingleFile

from PIL import Image

model_loader_configs = [
    # These configs are provided for detecting model type automatically.
    # The format is (state_dict_keys_hash, state_dict_keys_hash_with_shape, model_names, model_classes, model_resource)
    (None, "9269f8db9040a9d860eaca435be61814", ["wan_video_dit"], [WanModel], "civitai"),
    (None, "aafcfd9672c3a2456dc46e1cb6e52c70", ["wan_video_dit"], [WanModel], "civitai"),
    (None, "6bfcfb3b342cb286ce886889d519a77e", ["wan_video_dit"], [WanModel], "civitai"),
    (None, "6d6ccde6845b95ad9114ab993d917893", ["wan_video_dit"], [WanModel], "civitai"),
    (None, "6bfcfb3b342cb286ce886889d519a77e", ["wan_video_dit"], [WanModel], "civitai"),
    (None, "349723183fc063b2bfc10bb2835cf677", ["wan_video_dit"], [WanModel], "civitai"),
    (None, "efa44cddf936c70abd0ea28b6cbe946c", ["wan_video_dit"], [WanModel], "civitai"),
    (None, "3ef3b1f8e1dab83d5b71fd7b617f859f", ["wan_video_dit"], [WanModel], "civitai"),
    (None, "a61453409b67cd3246cf0c3bebad47ba", ["wan_video_dit", "wan_video_vace"], [WanModel, VaceWanModel], "civitai"),
    (None, "cb104773c6c2cb6df4f9529ad5c60d0b", ["wan_video_dit"], [WanModel], "diffusers"),
    (None, "9c8818c2cbea55eca56c7b447df170da", ["wan_video_text_encoder"], [WanTextEncoder], "civitai"),
    (None, "5941c53e207d62f20f9025686193c40b", ["wan_video_image_encoder"], [WanImageEncoder], "civitai"),
    (None, "1378ea763357eea97acdef78e65d6d96", ["wan_video_vae"], [WanVideoVAE], "civitai"),
    (None, "ccc42284ea13e1ad04693284c7a09be6", ["wan_video_vae"], [WanVideoVAE], "civitai"),
    (None, "7a513e1f257a861512b1afd387a8ecd9", ["wan_video_dit", "wan_video_vace"], [WanModel, VaceWanModel], "civitai"),
]

def resize_and_crop_center(image: Image.Image, target_height: int, target_width: int) -> Image.Image:
    """Resize the image while maintaining aspect ratio and center crop to target size.

    Args:
        image (PIL.Image): Input image.
        target_height (int): Target height after processing.
        target_width (int): Target width after processing.

    Returns:
        PIL.Image: Resized and center-cropped image.
    """
    # Step 1: Resize while keeping aspect ratio, to cover target size
    original_width, original_height = image.size
    scale = max(target_width / original_width, target_height / original_height)
    new_width = round(original_width * scale)
    new_height = round(original_height * scale)
    image = image.resize((new_width, new_height), Image.LANCZOS)

    # Step 2: Center crop
    left = (new_width - target_width) // 2
    upper = (new_height - target_height) // 2
    right = left + target_width
    lower = upper + target_height
    image = image.crop((left, upper, right, lower))

    return image

def preprocess(input_image):
    if input_image.mode == "P":
        input_image = input_image.convert("RGBA")
        # convert RGBA image to RGB
        new_image = Image.new("RGBA", input_image.size, 'WHITE')  
        # Create a white rgba background
        new_image.paste(input_image, (0, 0), input_image)               
        # Paste the image on the background. Go to the links given below for details.
        input_image = new_image.convert('RGB')                
    elif input_image.mode == "RGBA":
        # convert RGBA image to RGB
        new_image = Image.new("RGBA", input_image.size, 'WHITE')  
        # Create a white rgba background
        new_image.paste(input_image, (0, 0), input_image)               
        # Paste the image on the background. Go to the links given below for details.
        input_image = new_image.convert('RGB')
    else:
        input_image = input_image.convert("RGB")
        
    return input_image

model_manager = ModelManager(torch_dtype=torch.bfloat16, device="cpu")
model_manager.model_detector = [
    ModelDetectorFromSingleFile(model_loader_configs),
    ModelDetectorFromHuggingfaceFolder([]),
    ModelDetectorFromPatchedSingleFile([]),
]
model_manager.load_models(
    [
        [
            "/data/yuxiong/pretrained-weights/Wan2.1-VACE-14B/diffusion_pytorch_model-00001-of-00007.safetensors",
            "/data/yuxiong/pretrained-weights/Wan2.1-VACE-14B/diffusion_pytorch_model-00002-of-00007.safetensors",
            "/data/yuxiong/pretrained-weights/Wan2.1-VACE-14B/diffusion_pytorch_model-00003-of-00007.safetensors",
            "/data/yuxiong/pretrained-weights/Wan2.1-VACE-14B/diffusion_pytorch_model-00004-of-00007.safetensors",
            "/data/yuxiong/pretrained-weights/Wan2.1-VACE-14B/diffusion_pytorch_model-00005-of-00007.safetensors",
            "/data/yuxiong/pretrained-weights/Wan2.1-VACE-14B/diffusion_pytorch_model-00006-of-00007.safetensors",
            "/data/yuxiong/pretrained-weights/Wan2.1-VACE-14B/diffusion_pytorch_model-00007-of-00007.safetensors",
        ],
        "/data/yuxiong/pretrained-weights/Wan2.1-VACE-14B/models_t5_umt5-xxl-enc-bf16.pth",
        "/data/yuxiong/pretrained-weights/Wan2.1-VACE-14B/Wan2.1_VAE.pth",
    ],
    torch_dtype=torch.bfloat16, # You can set `torch_dtype=torch.bfloat16` to disable FP8 quantization.
)
pipe = WanVideoPipeline.from_model_manager(model_manager, device="cuda")
pipe.enable_vram_management(num_persistent_param_in_dit=None)

image = Image.open("/data/yuxiong/DiffSynth-Studio/examples/wanvideo/baili_beach.jpg")
image = preprocess(image)
image = resize_and_crop_center(image, target_height=832, target_width=480)
image.save("tmp_debug.png")

vace_video = VideoData("/data/yuxiong/projects/VACE/processed/frameref/2025-05-20-06-13-01/src_video-frameref.mp4", height=832, width=480)
vace_mask = VideoData("/data/yuxiong/projects/VACE/processed/frameref/2025-05-20-06-13-01/src_mask-frameref.mp4", height=832, width=480)
video = pipe(
    prompt="",
    negative_prompt="色调艳丽，过曝，静态，细节模糊不清，字幕，风格，作品，画作，画面，静止，整体发灰，最差质量，低质量，JPEG压缩残留，丑陋的，残缺的，多余的手指，画得不好的手部，画得不好的脸部，畸形的，毁容的，形态畸形的肢体，手指融合，静止不动的画面，杂乱的背景，三条腿，背景人很多，倒着走",
    num_inference_steps=25,
    height=832,
    width=480,
    num_frames=81,
    vace_video=vace_video,
    vace_video_mask=vace_mask,
    seed=112312, tiled=True,
    cfg_scale=5.0,
    sigma_shift=16.0,
)
save_video(video, "/data/yuxiong/DiffSynth-Studio/examples/wanvideo/beach.mp4", fps=16, quality=5)`

I tried the exactly same setting in Offical Gradio but getting good result

**Comment by matabear-wyx** [comment]
I assume the reason is inconsistent implementation of NoiseScheduler, I will try to implement FlowUniPCMultistepScheduler.

**Comment by UknowSth** [comment]
I encountered the same issue. Is it only possible to train using `FlowUniPCMultistepScheduler` ? Should the training_target be set as `noise - latents`? I noticed that the `FlowUniPCMultistepScheduler` in the Wan2.1 GitHub repository doesn't have this function.

**Comment by matabear-wyx** [comment]
> I assume the reason is inconsistent implementation of NoiseScheduler, I will try to implement FlowUniPCMultistepScheduler.

Update, diffusers has a correct version of VACE implementation and the results successfully align with official implementation so I'm not going to contribute this.
